src/components/controls/link/km-add-link.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <button type="button" class="btn btn-primary" @click="addLink()">
+    <slot name="link-button-label">+ Add Link</slot>
+  </button>  
+  <link-editor ref="linkEditorRef" @on-close="closed($event)">
+    <template v-slot:modalTitle>
+      <slot name="link-dialog-title"   >        
+        Editing link    
+      </slot>
+     </template>       
+  </link-editor>
+  <slot name="links-view">
+    <km-view-links v-model="links"   @on-delete = "remove($event)"   @on-edit = "edit($event)"></km-view-links>  
+  </slot>
+</template>
+<script setup>
+  import { ref } from 'vue'  
+  import kmViewLinks from './km-view-links.vue';
+  import linkEditor  from './link-editor.vue';
+
+  const linkEditorRef= ref(null); 
+  let editedLinkIndex = -1;
+  const links =ref([]);
+
+    function addLink() {  
+        edit(-1);
+    }      
+
+    function edit(index) {  
+        editedLinkIndex = index;  
+        linkEditorRef.value.show(links.value[index] ||{})     
+    }
+
+    function remove(index) {  
+        links.value.splice(index, 1);   
+    }
+
+    function closed(newValue) {    
+        if(Object.keys(newValue).length ==0) {// mean cacnel => return              
+            return;
+        } 
+        else{
+            if(editedLinkIndex<0) {  
+                      
+                links.value.push(newValue);
+            } 
+            else {        
+                links.value[editedLinkIndex]=newValue;
+            } 
+            editedLinkIndex = -1;
+        }       
+    }
+</script>
```

Comment:
there is no model on the component, how do you plan to send data back to the parent and receive existing data